### PR TITLE
Add vqfx inventory

### PR DESCRIPTION
**Pull request by danielmellado**
This commit modifies the template for junos ansible testing so it
includes also vqfx as an appliance.

### CODE REVIEW

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 15371639631d1aa6239623437d7efe9536a9f3d2

File: playbooks/ansible-test-network-integration-base/templates/inventory-junos.j2 [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #  DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
 #
 [junos]
+vqfx-18.1R3 ansible_host={{ hostvars['vqfx-18.1R3'].ansible_host }} ansible_user=zuul
```
Comment (pabelanger) [inline playbooks/ansible-test-network-integration-base/templates/inventory-junos.j2]: So, this won't work, as each nodeset doesn't know of the other. So we need to see if the host exists in hostvars, or do something like hostvars[appliance[0]]

To get the first host in the appliance group

Comment (pabelanger) [reply playbooks/ansible-test-network-integration-base/templates/inventory-junos.j2]: The think to remember, hostvars is passed from outer ansible (zuul-executor) into inner zuul

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now dfff17cb580ad0ae5eabfe8124d8d8307b21e695

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now f14d44251b156a34ccc0af5765a25a7216f9cdf5

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now a82ed56c810347fc0ac8a08da3c5b642a34c862b

File: zuul.d/project-templates.yaml [hunk 2]
```diff
@@ -816,6 +854,134 @@
...
+              - ^lib/ansible/plugins/terminal/__init__.py
+              - ^lib/ansible/plugins/terminal/junos.py
+              - ^test/integration/targets/junos_.*
+              - ^test/integration/targets/netconf_.*
+              - ^test/integration/targets/prepare_junos_tests/.*
+            voting: false
+        - ansible-test-network-integration-junos-vqfx-python37:
+            branches:
+              - devel
+              - stable-2.9
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml]: There is no need to add to stable branches here, as we'll only start testing it against devel

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now eae45abb0333cf6449a8c692e931aeaac59b21a5

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now e7540d971aa7b1074a1cf2de725c79c707d01a57

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 99c973b22417e396e41a92881665f6edc4e1011f

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now c784fdeebdaf29517c27f35f25673788af83ed4b

File: zuul.d/jobs.yaml [hunk 3]
```diff
@@ -384,6 +384,91 @@
     vars:
       ansible_test_integration_targets: "netconf_.*"
 
+- job:
+    name: ansible-network-junos-appliance
+    parent: ansible-network-appliance-base
+    pre-run: playbooks/ansible-network-junos-appliance/pre.yaml
+    run: playbooks/ansible-network-junos-appliance/run.yaml
+    abstract: true
```
Comment (pabelanger) [inline zuul.d/jobs.yaml]: This job is duplicated, and should be deleted

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now bd53737018cfb2c9ef11a5ceb63a5a16ff09a61f

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 58f917bac324b0ea27624122163f3f55eb8846e0

File: playbooks/ansible-network-appliance-base/pre.yaml [hunk 4]
```diff
@@ -75,3 +75,13 @@
...
+      register: result
+
+    - name: Debug hostvars
+      debug:
+        var: result
+        verbosity: 4
+
+    - name: cat ansible_hosts to shell
+      shell: cat /etc/ansible/hosts
+      ignore_errors: true
```
Comment (pabelanger) [inline playbooks/ansible-network-appliance-base/pre.yaml]: lets delete this now

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 09ae24c03bbfb36d0aa901dbbe715ef4368e1b1a

File: playbooks/ansible-network-appliance-base/pre.yaml [hunk 5]
```diff
@@ -75,3 +75,4 @@
         write_inventory_dest: /etc/ansible/hosts
         write_inventory_exclude_hostvars:
           - ansible_connection
+      register: result
```
Comment (pabelanger) [inline playbooks/ansible-network-appliance-base/pre.yaml]: should be removed

File: zuul.d/project-templates.yaml [hunk 6]
```diff
@@ -867,11 +897,129 @@
...
+              - ^lib/ansible/plugins/terminal/junos.py
+              - ^test/integration/targets/junos_.*
+              - ^test/integration/targets/netconf_.*
+              - ^test/integration/targets/prepare_junos_tests/.*
+            voting: false
         - ansible-test-network-integration-junos-vsrx-python27:
             branches:
               - devel
-              - stable-2.9
-              - stable-2.8
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml]: please add back

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 4ad759de36872fbdb56b987a41097dd113657a70

File: zuul.d/jobs.yaml [hunk 7]
```diff
@@ -384,6 +384,92 @@
...
+    name: ansible-test-network-integration-junos-vqfx-python37
+    parent: ansible-test-network-integration-junos-vqfx
+    nodeset: vqfx-18.1R3-python37
+    vars:
+      ansible_test_python: 3.7
+
+- job:
+    name: ansible-test-network-integration-junos-vqfx-python38
+    parent: ansible-test-network-integration-junos-vqfx
+    nodeset: vqfx-18.1R3-python36
```
Comment (pabelanger) [inline zuul.d/jobs.yaml]: this should be python38

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now d859caf27f163a48b86fb8c03654a4759e332c64

**Pull request closed without merging**